script.js

```diff
@@ -0,0 +1,360 @@
+// For example, for [1, 2, 2] it should return 9.
+
+function getSquareSum(arrayOfNumbers) {
+    let squaredSum = 0;
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+        const squaredNumber = arrayOfNumbers[i] ** 2;
+        squaredSum = squaredSum + squaredNumber;
+    }
+    return squaredSum;
+}
+
+console.log(getSquareSum([1, 2, 2]));
+
+// 10. Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.
+// The output should be two capital letters with a dot separating them.
+// It should look like this:
+// Sam Harris => S.H
+// patrick feeney => P.F
+
+function getInitials(name) {
+    const splitName = name.split(' ');
+    let initials = splitName[0].substring(0, 1).toUpperCase();
+    if (splitName.length > 1) {
+        initials += '.' + splitName[splitName.length - 1].substring(0, 1).toUpperCase();
+    }
+    return initials;
+}
+
+// Example usage
+console.log(getInitials("Sam Harris")); // Output: "S.H"
+console.log(getInitials("patrick feeney")); // Output: "P.F"
+
+// 11. Consider an array/list of sheep where some sheep may be missing from their place.
+// We need a function that counts the number of sheep present in the array (true means present).
+
+function getCountSheeps(sheep) {
+    let numberOfSheeps = 0;
+    for (let i = 0; i < sheep.length; i++) {
+        if (sheep[i] === true) {
+            numberOfSheeps++;
+        }
+    }
+    return numberOfSheeps;
+}
+
+console.log(getCountSheeps(
+    [true,  true,  true,  false,
+    true,  true,  true,  true ,
+    true,  false, true,  false,
+    true,  false, false, true ,
+    true,  true,  true,  true ,
+    false, false, true,  true]));
+
+// 12. Convert number to reversed array of digits
+// Given a random non-negative number, you have to return the digits of this number within an array in reverse order.
+
+function getReversedArrayOfDigits(numbers) {
+    const convertedToString = numbers.toString();
+    const splitString = convertedToString.split('');
+    const reverseString = splitString.reverse();
+    return reverseString.map(Number);
+}
+
+console.log(getReversedArrayOfDigits(35231));
+
+// 13. Write a function findNeedle() that takes an array full of junk but containing one "needle".
+
+function findNeedle(haystack) {
+    for (let i = 0; i < haystack.length; i++) {
+        if (haystack[i] === 'needle') {
+            return 'found the needle at position ' + i;
+        }
+    }
+}
+
+console.log(findNeedle(["hay", "junk", "hay", "hay", "moreJunk", "needle", "randomJunk"]));
+
+//  14. Given an array of integers, return a new array with each value doubled.
+// For example:
+// [1, 2, 3] --> [2, 4, 6]
+
+function getDoubledArray(arrayOfNumbers){
+    let doubledArray = [];
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+        const duplicatedValue = arrayOfNumbers[i] + arrayOfNumbers[i];
```

Comment:
`duplicated` is not the same as `doubled`

https://diffsense.com/diff/double/duplicate